proposals/0008-swift-prometheus.md

```diff
@@ -0,0 +1,158 @@
+* Histogram - A  *histogram*  samples observations (usually things like request durations or response sizes) and counts them in configurable buckets. It also provides a sum of all observed values.
+* Summary - Similar to a  *histogram* , a  *summary*  samples observations (usually things like request durations and response sizes). While it also provides a total count of observations and a sum of all observed values, it calculates configurable quantiles over a sliding time window.
+
+SwiftPrometheus provides fully featured implementations for all of them, including a thin wrapper around them for integration with `swift-metrics`.
+
+### API Layout
+Below section will lay out the public API of this package. For the internal APIs I would suggest you to read through the code on GitHub :smile:. This section is split up into two parts, using it integrated with the `swift-metrics` package or using this library standalone.
+
+#### With swift-metrics
+For use with swift-metrics, bootstrap the MetricsSystem with an instance of `PrometheusClient`:
+```swift
+let myProm = PrometheusClient()
+MetricsSystem.bootstrap(myProm)
+```
+After that, you can use the metric types used by swift-metrics for your metrics. The mapping is as follows:
+| swift-metrics | SwiftPrometheus |
+|--|--|
+| Counter | Counter |
+| Gauge | Gauge |
+| Recorder (agg) | Histogram |
+| Timer | Summary |
+
+---
+
+To get a hold of your `PrometheusClient` either to:
+a) use custom prometheus behaviour; or
+b) get your metrics output
+there is a utility function on `MetricsSystem`
+```swift
+let myProm = try MetricsSystem.prometheus()
+```
+This will either return the `PrometheusClient` used with `.bootstrap()` or throw an error if `MetricsSystem` was not bootstrapped with `PrometheusClient`
+*Note: There currently is no support for retrieving `PrometheusClient` when being used with `MultiplexMetricsHandler`* 
+
+#### Without swift-metrics
+
+To get started, initialise an instance of `PrometheusClient`
+```swift
+import Prometheus
+let myProm = PrometheusClient()
+```
+Once done, you can use the `create*` APIs to create any of the above described metric types.
+
+```swift
+// MetricLabels is a helper type used to add labeled metrics.
+struct MyCodable: MetricLabels {
+   var thing: String = "*"
+}
+
+// - Counter
+let counter = myProm.createCounter(forType: Int.self, named: "my_counter", helpText: "Just a counter", initialValue: 42, withLabelType: MyCodable.self)
+
+counter.inc() // Increment by one
+counter.inc(12) // Increment by a value
+counter.inc(12, MyCodable(thing: "test")) // Increment a labeled counter
+
+// - Gauge
+let gauge = myProm.createGauge(forType: Int.self, named: "my_gauge", helpText: "Just a gauge", initialValue: 42, withLabelType: MyCodable.self)
+gauge.inc() // Same APIs as Counter
+gauge.dec() // Same APIs as `inc()` but reversed.
+gauge.set(42) // Set the gauge to a specific value
+
+// - Histogram
+// Histograms use special labels, different than the Counter & Gauge
```

Comment:
```suggestion
// Histograms in Prometheus use special labels, which are different from Counter & Gauge
```

Reply:
The context should be clear here with regards to the rest of the code block

Reply:
Works for me!